src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1252,8 +1252,7 @@ static void uct_ib_mlx5_devx_cleanup_flush_mr(uct_ib_mlx5_md_t *md)
 {
     ucs_status_t status;
 
-    if (!(md->flags & UCT_IB_MLX5_MD_FLAG_KSM) ||
-        !uct_ib_md_is_flush_rkey_valid(md->super.flush_rkey)) {
+    if (!uct_ib_md_is_flush_rkey_valid(md->super.flush_rkey)) {
```

Comment:
if KSM flag is off, flush_mr should also not exist. do you see otherwise?

Reply:
I see that flush_mr and key is created successfully before atomic_mr creation fails then KSM flag is dropped.

Reply:
hmm i see so we really need to check the flush key and not MD flag..